CHANGES.rst

```diff
@@ -1,15 +1,19 @@
 Changes
 =======
 
-Version 0.9.1 (UNRELEASED)
+Version 0.9.1 (2023-09-27)
 --------------------------
```

Comment:
We should add:

- Fixes container image names to be Podman-compatible.

Also should we add one entry for https://github.com/reanahub/reana-job-controller/pull/400 ?

Reply:
Added, and expanded an existing item about Rucio sidecar to explicitly hint that users can override it.